frontend/src/app/specification-overview/specification-overview.component.ts

```diff
@@ -38,6 +43,38 @@ export class SpecificationOverviewComponent implements OnInit {
     }
   }
 
+  public async downloadSpecification(specification: Specification, fileType: string) {
+    const latestVersion = specification.versions[0];
+    if (latestVersion !== undefined) {
+      this.downloadVersion[fileType](specification, latestVersion);
+    }
+  }
+
+  private createDownloadFileName(specification: Specification, version: Version) {
```

Comment:
Regarding ordering: I always like to structure code beginning with the most abstract methods and going deeper further down. Each method is beneath the one that calls it (if there's only one caller), or maybe beneath all of the callers.